boards: stm32f769i_disc0: add touchscreen support

papadkostas opened the pull request:
Add support for the touchscreen controller on STM32F769I-DISC0(MB1166).
The touchscreen uses Focaltech FT6202 which is compatible with FT5336.
Tested with kscan_touch sample application.

Signed-off-by: Konstantinos Papadopoulos <kostas.papadopulos@gmail.com>

papadkostas commented:
devicetree error: Label 'touch_controller' appears on /soc/i2c@40005400/ft5336@38 and on /soc/i2c@40006000/ft6202@2a
Should I just rename it on the board dts or is there a way to skip a test for particular boards? Since it doesn't make sense to put adafruit display board on f769 discovery.

erwango commented:
> devicetree error: Label 'touch_controller' appears on /soc/i2c@40005400/ft5336@38 and on /soc/i2c@40006000/ft6202@2a Should I just rename it on the board dts or is there a way to skip a test for particular boards? Since it doesn't make sense to put adafruit display board on f769 discovery.

You can add a `platform_exclude` to this particular sample test variant:
```suggestion
  sample.filesystem.fat_fs.adafruit_2_8_tft_touch_v2:
     depends_on: arduino_spi arduino_gpio arduino_i2c
+    platform_exclude: stm32f769i_disc0
     extra_args: SHIELD=adafruit_2_8_tft_touch_v2
```

erwango commented on boards/arm/stm32f769i_disco/Kconfig.defconfig:
```diff
@@ -12,6 +12,16 @@ config SPI_STM32_INTERRUPT
+config I2C
+	default y if KSCAN
+
+if KSCAN
+
+config KSCAN_FT5336
+	default y
+
+endif # KSCAN
```
```suggestion
config KSCAN_FT5336
	default y if KSCAN
```

papadkostas replied:
@erwango Change done. Should I make another commit to exclude the board from this sample?.

erwango replied:
> @erwango Change done. 

Thanks

> Should I make another commit to exclude the board from this sample?.

Yes please. Though please take care to fix commit description: 
CI is failing due to incomplete commit message. It should looks like:
```
<path>: <component>: Title

Commit description.

Signed-off-by: <name> <email>
```
Cf https://docs.zephyrproject.org/latest/contribute/index.html#commit-guidelines

papadkostas replied:
Fixed the commit message and also opened a PR for the samples.

erwango reviewed (changes requested):
Nitpick, otherwise looks good

papadkostas committed:
boards: stm32f769i_disc0: add touchscreen support

Add support for the touchscreen controller on STM32F769I-DISC0(MB1166).
The touchscreen uses Focaltech FT6202 which is compatible with FT5336.
Tested with kscan_touch sample application.

Signed-off-by: Konstantinos Papadopoulos <kostas.papadopulos@gmail.com>

erwango commented:
@papadkostas Can you rebase now that the other PR was merged ?

MaureenHelm reviewed (changes requested):
Code changes look fine, but please rebase and remove the merge commit
